File: exercise/day02/src/test/java/games/FizzBuzzTests.java
package games;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;
import static org.assertj.core.api.Assertions.assertThatThrownBy;

class FizzBuzzTests {
    @Test
    void returns_the_given_number_for_1() throws OutOfRangeException {
        assertThat(FizzBuzz.convert(1))
                .isEqualTo("1");
    }

    @Test
    void returns_the_given_number_for_67() throws OutOfRangeException {
        assertThat(FizzBuzz.convert(67))
                .isEqualTo("67");
    }

    @Test
    void returns_the_given_number_for_82() throws OutOfRangeException {
        assertThat(FizzBuzz.convert(82))
                .isEqualTo("82");
    }

    @Test
    void returns_Fizz_for_3() throws OutOfRangeException {
        assertThat(FizzBuzz.convert(3))
                .isEqualTo("Fizz");
    }

    @Test
    void returns_Fizz_for_66() throws OutOfRangeException {
        assertThat(FizzBuzz.convert(66))
                .isEqualTo("Fizz");
    }

    @Test
    void returns_Fizz_for_99() throws OutOfRangeException {
        assertThat(FizzBuzz.convert(99))
                .isEqualTo("Fizz");
    }

    @Test
    void returns_Buzz_for_5() throws OutOfRangeException {
        assertThat(FizzBuzz.convert(5))
                .isEqualTo("Buzz");
    }

    @Test
    void returns_Buzz_for_50() throws OutOfRangeException {
        assertThat(FizzBuzz.convert(50))
                .isEqualTo("Buzz");
    }

    @Test
    void returns_Buzz_for_85() throws OutOfRangeException {
        assertThat(FizzBuzz.convert(85))
                .isEqualTo("Buzz");
    }

    @Test
    void returns_FizzBuzz_for_15() throws OutOfRangeException {
        assertThat(FizzBuzz.convert(15))
                .isEqualTo("FizzBuzz");
    }

    @Test
    void returns_FizzBuzz_for_30() throws OutOfRangeException {
        assertThat(FizzBuzz.convert(30))
                .isEqualTo("FizzBuzz");
    }

    @Test
    void returns_FizzBuzz_for_45() throws OutOfRangeException {
        assertThat(FizzBuzz.convert(45))
                .isEqualTo("FizzBuzz");
    }

    @Test
    void throws_an_exception_for_0() {
        assertThatThrownBy(() -> FizzBuzz.convert(0))
                .isInstanceOf(OutOfRangeException.class);
    }

    @Test
    void throws_an_exception_for_101() {
        assertThatThrownBy(() -> FizzBuzz.convert(101))
                .isInstanceOf(OutOfRangeException.class);
    }

    @Test
    void throws_an_exception_for_minus_1() {
        assertThatThrownBy(() -> FizzBuzz.convert(-1))
                .isInstanceOf(OutOfRangeException.class);
    }
}

File: exercise/c#/Day06/Day06.Tests/FizzBuzzTests.cs
using FluentAssertions;
using Xunit;

namespace Day06.Tests
{
    public class FizzBuzzTests
    {
        #region "Normal" numbers

        [Fact]
        public void Returns_The_Given_Number_For_1() => FizzBuzz.Convert(1).Should().Be("1");

        [Fact]
        public void Returns_The_Given_Number_For_67() => FizzBuzz.Convert(67).Should().Be("67");

        [Fact]
        public void Returns_The_Given_Number_For_82() => FizzBuzz.Convert(82).Should().Be("82");

        #endregion

        #region Fizz

        [Fact]
        public void Returns_Fizz_For_3() => FizzBuzz.Convert(3).Should().Be("Fizz");

        [Fact]
        public void Returns_Fizz_For_66() => FizzBuzz.Convert(66).Should().Be("Fizz");

        [Fact]
        public void Returns_Fizz_For_99() => FizzBuzz.Convert(99).Should().Be("Fizz");

        #endregion

        #region Buzz

        [Fact]
        public void Returns_Buzz_For_5() => FizzBuzz.Convert(5).Should().Be("Buzz");

        [Fact]
        public void Returns_Buzz_For_50() => FizzBuzz.Convert(50).Should().Be("Buzz");

        [Fact]
        public void Returns_Buzz_For_85() => FizzBuzz.Convert(85).Should().Be("Buzz");

        #endregion

        #region FizzBuzz

        [Fact]
        public void Returns_FizzBuzz_For_15() => FizzBuzz.Convert(15).Should().Be("FizzBuzz");

        [Fact]
        public void Returns_FizzBuzz_For_30() => FizzBuzz.Convert(30).Should().Be("FizzBuzz");

        [Fact]
        public void Returns_FizzBuzz_For_45() => FizzBuzz.Convert(45).Should().Be("FizzBuzz");

        #endregion

        #region Failures

        [Fact]
        public void Throws_An_Exception_For_0()
        {
            var act = () => FizzBuzz.Convert(0);
            act.Should().Throw<OutOfRangeException>();
        }

        [Fact]
        public void Throws_An_Exception_For_101()
        {
            var act = () => FizzBuzz.Convert(101);
            act.Should().Throw<OutOfRangeException>();
        }

        [Fact]
        public void Throws_An_Exception_For_Minus_1()
        {
            var act = () => FizzBuzz.Convert(-1);
            act.Should().Throw<OutOfRangeException>();
        }

        #endregion
    }
}